src/dmd/root/ctfloat.d

```diff
@@ -39,28 +39,20 @@ private
 extern (C++) struct CTFloat
 {
   nothrow:
-    version(DigitalMars)
-    {
-        static __gshared bool yl2x_supported = true;
-        static __gshared bool yl2xp1_supported = true;
-    }
-    else
-    {
-        static __gshared bool yl2x_supported = false;
-        static __gshared bool yl2xp1_supported = false;
-    }
+    enum yl2x_supported = __traits(compiles, core.math.yl2x(1.0L, 2.0L));
```

Comment:
The declaration exists in core.math unconditionally, so it always compiles. For DMD (and LDC?), it is converted to an intrinsic. I think it should be excluded in builtin_init for appropriate versions. This would also produce better error messages for GDC instead of asserting here.

Reply:
For DMD it's a builtin and always declared.
For LDC, it's a regular inline asm function, defined for x86[_64] targets using 80-bit `real` only, and only since a couple of versions.
For GDC, it's apparently just a declaration without implementation => linker error.

So one possibility is:
````D
version(GNU)
    enum yl2x_supported = false;
else
    enum yl2x_supported = __traits(compiles, core.math.yl2x(1.0L, 2.0L));
enum yl2xp1_supported = yl2x_supported; // most likely redundant
```

[This automatically makes sure the CTFloat functions aren't called, so no asserts for GDC.]